aiidalab_widgets_base/__init__.py

```diff
@@ -1,7 +1,39 @@
 """Reusable widgets for AiiDAlab applications."""
+import os
+
+from aiida.manage import get_profile
+
+_WARNING_TEMPLATE = """
+<div style="background-color: #f7f7f7; border: 2px solid #e0e0e0; padding: 20px; border-radius: 5px;">
+    <p style="font-size: 16px; font-weight: bold; color: #ff5733;">Warning:</p>
+    <p style="font-size: 14px;">The default profile '<span style="font-style: italic;">{profile}</span>' was loaded automatically. This behavior will be removed in the <span style="font-style: italic;">{version}</span>. Please load the profile manually before loading modules from aiidalab-widgets-base by adding the following code at the beginning cell of the notebook:</p>
+    <pre style="background-color: #f0f0f0; padding: 10px; border: 1px solid #ccc; font-family: 'Courier New', monospace;">
 from aiida import load_profile
+load_profile();</pre>
+</div>
+"""
+
+# use the environment variable to control the automatic profile loading
+# so it can be disabled for testing, e.g. in the CI
+# use `TESTING=1 pytest` to disable the automatic profile loading
+TESTING = os.environ.get("TESTING", "False").lower() in ["true", "1"]
```

Comment:
I am not a fan of this solution. Could we instead detect whether we run in the Jupyter notebook context (perhaps checking whether ipython is loaded or something like that?). That way we should be able to detect when we're unit testing and not load the profile.

Reply:
I also adapt suggestions from @yakutovicha, the import is with module namespaces now.

Reply:
Great! I like the solution with `is_running_in_jupyter`. :+1: